### PR TITLE
[ENH] Complete method

### PR DESCRIPTION
**Pull request by samukweku**
# PR Description

Please describe the changes proposed in the pull request:

-  Fixed wrong outputs
- Use `pd.merge` to handle duplicate  data and maintain column positions
-

**This PR improves #727 .**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [x] Add a line to `CHANGELOG.rst` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

## Quick Check

To do a very quick check that everything is correct, follow these steps below:

- [x] Run the command `make check` from pyjanitor's top-level directory. This will automatically run:
    - black formatting
    - flake8 checking
    - running the test suite
    - docs build

Once done, please check off the check-box above.

If `make check` does not work for you, you can execute the commands listed in the Makefile individually.

## Code Changes

If you are adding code changes, please ensure the following:

- [x] Ensure that you have added tests.
- [x] Run all tests (`$ pytest .`) locally on your machine.
    - [x] Check to ensure that test coverage covers the lines of code that you have added.
    - [x] Ensure that all tests pass.

## Documentation Changes

If you are adding documentation changes, please ensure the following:

- [x] Build the docs locally.
- [x] View the docs to check that it renders correctly.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: tests/functions/test_complete.py [hunk 1]
```diff
@@ -73,8 +24,92 @@ def df1():
     )
 
 
+def test_empty_column(df1):
+    "Return dataframe if `columns` is empty."
```
Comment (ericmjl) [inline tests/functions/test_complete.py:26]: Docstring! (Just a friendly reminder :smile:.)

File: tests/functions/test_complete.py [hunk 2]
```diff
@@ -136,81 +167,163 @@ def test_fill_value_all_years(df1):
...
+            "Taxon",
+        ],
+        fill_value={"Abundance": 0},
+    )
+    assert_frame_equal(result, df1_output)
 
 
-frame = pd.DataFrame(
+# from https://tidyr.tidyverse.org/reference/complete.html
+df = pd.DataFrame(
```
Comment (ericmjl) [inline tests/functions/test_complete.py]: Do you intend for this dataframe to be a fixture instead?

**Review by ericmjl — CHANGES_REQUESTED** [review]
Just two general comments re: the tests docstrings. Nothing else beyond that! Much appreciated, @samukweku!

File: tests/functions/test_complete.py (hunk 2 above)
Comment (samukweku) [reply tests/functions/test_complete.py]: I used it in a parametrization. Reordered the data though; hopefully, it is clearer.

File: janitor/utils.py [hunk 3]
```diff
@@ -539,69 +539,140 @@ def _grid_computation(entry: Dict) -> pd.DataFrame:
...
-    for item in list_of_columns:
-        if not isinstance(item, (str, dict, list, tuple)):
-            raise ValueError(
-                """Value must either be a column label, a list/tuple of columns or a
-                    dictionary where the keys are columns in the dataframe."""
-            )
-        if not item:
+# copied from pandas.core.common.py
+# used in _computations_complete
+# might also be useful in case_when
```
Comment (ericmjl) [inline janitor/utils.py]: @samukweku if this one is copied from the pandas' core, could we import it instead of copy/pasting it? Or is it modified?

**Review by ericmjl — APPROVED** [review]
Approving! The comment below, @samukweku, isn't so important, though being aware of minimizing code duplication is a good thing.

File: janitor/utils.py (hunk 3 above)
Comment (samukweku) [reply janitor/utils.py]: Thanks for the suggestion. Easier to import.

**Comment by ericmjl** [comment]
@samukweku just the conflicts to resolve, and we're good to merge! Please feel free to do the honours; the PR has been open for a long time indeed.

**Pull request merged**